### DISCUSSION TITLE
Pretendard 글꼴에 일본어, 한자도 포함되어 있는걸까요?

### DISCUSSION THREAD

**Discussion by dungsil** [orioncactus/pretendard#86] [Q&A] [answered]
안녕하세요
우선 좋은 글꼴 만들어 주셔서 감사합니다.

한국어, 영어. 일본어를 지원할 때 Pretendard만 사용하면 되는 지 Pretendard JP와 Pretendard를 같이 사용해야 하는지 햇갈려서 문의드립니다

**Comment by orioncactus** [comment, marked as answer]
안녕하세요! 늦게 답장드려 죄송합니다.

Pretendard JP에는 기존 Pretendard에 일본 한자도 추가되어 있습니다. 따라서 앱에서 사용하시는 목적이면 Pretendard JP만 사용하셔도 됩니다. 웹에서 용량을 최적화하는 것을 고려하신다면 dynamic-subset으로 제공하는 코드를 다음과 같이 사용하시는 것을 추천드리며, dynamic-subset으로 사용할 때는 CSS에서 font-family를 주로 쓸 언어 순서대로 선언하시면 됩니다. 주로 사용하는 언어가 일본어, 한국어, 영어 순일 때 `font-family: "Pretendard JP", Pretendard, "Pretendard Std", ...`와 같이 적는 식입니다.

#### HTML
```HTML
<link rel="preload" as="style" crossorigin="anonymous" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.3/dist/web/static/pretendard-jp-dynamic-subset.css">
<link rel="preload" as="style" crossorigin="anonymous" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.3/dist/web/static/pretendard-dynamic-subset.css">
<link rel="preload" as="style" crossorigin="anonymous" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.3/dist/web/static/pretendard-std-dynamic-subset.css">
<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.3/dist/web/static/pretendard-jp-dynamic-subset.css">
<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.3/dist/web/static/pretendard-dynamic-subset.css">
<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard@v1.3.3/dist/web/static/pretendard-std-dynamic-subset.css">
```

사용하시는 데 어려움이 있으시다면 언제든 알려주세요!
감사합니다.

**Reply by dungsil** [reply]
혹시, 그러면 일본어/한국어 서브셋만 따로 제작해주실 계획은 없으실까요..?

**Reply by orioncactus** [reply]
[fontTools pyftsubset](https://fonttools.readthedocs.io/en/latest/subset/index.html)을 이용해 원하는 문자 세트로 서브셋을 제작하실 수 있습니다. 어떤 부분에서 일본어/한국어 서브셋을 만드려고 하시는지 알 수 있을까요?
